### ISSUE TITLE
Move away from MultiMC metadata services

### ISSUE THREAD

**Issue by d-513** [PolyMC/PolyMC#75]
This is very important, since project can get sabotaged by multimc metadata, and also this will allow to work on better ARM support

**Comment by ghost** [comment]
The documentation for MultiMC's meta scripts is non-existent, so I need to figure that out first.

**Comment by DioEgizio** [comment]
Since it seems that multimc just blocked the launcher from accessing their meta server, we should at least mirror this https://github.com/MultiMC/meta-multimc

**Comment by d-513** [comment]
The PolyMC User agent is blocked.
```
$ curl -A "PolyMC/1.0.5" https://meta.multimc.org/v1/index.json
error code: 1020
$ curl -A "Chrome" https://meta.multimc.org/v1/index.json
{ actual json output }
```

**Comment by peterix** [comment]
The metadata is an internal detail of the launcher and not a public resource.
Please maintain your own.

**Comment by d-513** [comment]
bruh moment

**Comment by ghost** [comment]
> The metadata is an internal detail of the launcher and not a public resource. Please maintain your own.

I would love to, but there aren't any docs for it whatsoever.

**Comment by Scrumplex** [comment]
> The metadata is an internal detail of the launcher and not a public resource. Please maintain your own.

Thanks for this formal message. But communicating this only after cutting this project off is just hostile..

**Comment by peterix** [comment]
Yes.

**Comment by peterix** [comment]
And I will do it with any resource you abuse.

**Comment by ghost** [comment]
> And I will do it with any resource you abuse.

If you don't want us to abuse it then maybe document the metadata scripts?

**Comment by phit** [comment]
they are super simple scripts just read them, the config file is self explanatory too

**Comment by peterix** [comment]
I am not going to spend any effort on contributing to this mess.

**Comment by Scrumplex** [comment]
> And I will do it with any resource you abuse.

It's your right to determine what's rightful use and what's abuse, especially as it seems like you are hosting it on your own server. Again, a notice would be the right way, than cutting this project off. In fact that's industry standard when it comes to abuse of network resources.

**Comment by peterix** [comment]
You just want me to keep doing the work for you, while you leech off of it.

YEET

**Comment by ghost** [comment]
> YEET

Damn that was so cool I bet the kids think you're so cool

**Comment by Scrumplex** [comment]
off-topic: this thread is already a candidate for https://github.com/nikolas/github-drama

**Comment by Forkk** [comment]
Considering the lead maintainer of this project is the one who sabotaged our AUR package, and considering I've now had to deal with several discord raids from people linking to this project, expecting anything but hostility in return at this point seems a bit much to me.

As for meta, the "industry standard" is to host your own infrastructure and pay for your own bandwidth.

**Comment by Scrumplex** [comment]
If network resources are a concern, I can recommend looking at GitHub Pages. Makes a FOSS project less reliant on single points of failures as well

**Comment by ghost** [comment]
> Considering the lead maintainer of this project is the one who sabotaged our AUR package

If you don't want your AUR package to get taken down then maybe follow the package guidelines.

**Comment by ghost** [comment]
> off-topic: this thread is already a candidate for https://github.com/nikolas/github-drama

Submit it, this is kinda funny

**Comment by d-513** [comment]
I was  blocked lmao

**Comment by ghost** [comment]
> they are super simple scripts just read them, the config file is self explanatory too

It would be helpful if the scripts actually worked

**Comment by Forkk** [comment]
> > they are super simple scripts just read them, the config file is self explanatory too
> 
> It would be helpful if the scripts actually worked

If you're going to maintain this, maybe you should make sure you have the skills to do so. That would include doing your own work on meta. Good luck! :)

**Comment by peterix** [comment]
I believe this is a teachable moment.

You can expect this from every single dependency. I've been screwed over by most of them, some multiple times. The meta repo is a shield against that, and needs to be quickly adapted to third parties breaking compatibility, or becoming unavailable/hostile.

I do not have the time or inclination to work on more than one launcher, and that will always be MultiMC. Don't expect me to maintain stuff for you or to consider your needs. I care about *MultiMC* users, not users of any other launcher or tool. This extends to the meta repo.

I would like your thing to become cleanly separated. Especially because of your unfortunate choice to use the GPL. I cannot afford the MultiMC codebase becoming unusable in other projects because of that. There is only so much code I am willing to rewrite in my life.

In any case, you decided to do a hostile takeover, and when it blew up in your face, you did this. I don't care.

As for YEET, I like it. It expresses the sentiment perfectly.

**Comment by Jan200101** [comment]
> I believe this is a teachable moment.
> 
> You can expect this from every single dependency. I've been screwed over by most of them, some multiple times. The meta repo is a shield against that, and needs to be quickly adapted to third parties breaking compatibility, or becoming unavailable/hostile.
> 
> I do not have the time or inclination to work on more than one launcher, and that will always be MultiMC. Don't expect me to maintain stuff for you or to consider your needs. I care about _MultiMC_ users, not users of any other launcher or tool. This extends to the meta repo.

This is fair and I think no one was asking this of you.
The project should have gotten its own metadata service off the road as soon as it could.

> I would like your thing to become cleanly separated. Especially because of your unfortunate choice to use the GPL. I cannot afford the MultiMC codebase becoming unusable in other projects because of that. There is only so much code I am willing to rewrite in my life.
> 
> In any case, you decided to do a hostile takeover, and when it blew up in your face, you did this. I don't care.

Our views may differ on this but I am in the hopes of this trying to be rebranded version of Launcher that would function the same as MultiMC with the focus on distrobuting and maintaining features required for that (though how far that will go and how well it will keep up with upstream is to be seen).

On the blowing up part I think this should have come sooner, if this project seeks to be independent it should act like it.

> As for YEET, I like it. It expresses the sentiment perfectly.

**Comment by d-513** [comment]
Fixed in 1.0.6
